Auth0/Result.swift

```diff
@@ -24,11 +24,15 @@ import Foundation
 
 /**
  Result object for Auth0 APIs requests
-
- - Success: request completed successfuly with it's response body
- - Failure: request failed with a specific error
  */
-public enum Result<T> {
-    case success(result: T)
-    case failure(error: Error)
+public typealias Result<T> = Swift.Result<T, Error>
+
+extension Result {
```

Comment:
These shims can be removed by convert the `.success(result:)` to `.success()` and `.failure(error:)` to `.failure()` as done in https://github.com/auth0/Auth0.swift/compare/master...kiva:swift-result-type-no-shims